src/engine/shared/compression.cpp

```diff
@@ -7,20 +7,20 @@
 // Format: ESDDDDDD EDDDDDDD EDD... Extended, Data, Sign
 unsigned char *CVariableInt::Pack(unsigned char *pDst, int i)
 {
-	*pDst = (i>>25)&0x40; // set sign bit if i<0
-	i = i^(i>>31); // if(i<0) i = ~i
+	*pDst = (i >> 25) & 0x40; // set sign bit if i<0
+	i ^= i >> 31; // if(i<0) i = ~i
 
-	*pDst |= i&0x3F; // pack 6bit into dst
+	*pDst |= i & 0x3F; // pack 6bit into dst
 	i >>= 6; // discard 6 bits
 	if(i)
```

Comment:
While your on it: this part(L15-27) could be simplified.
-use a while here instead, the existing one and L23-25 could be removed then

Reply:
Fixed. I also fixed more undefined/implementation-specific behavior in L10-15, due to right shifting of negative numbers. Now it does what the comments already said.